Kconfig.zephyr

```diff
@@ -327,6 +327,15 @@ config NATIVE_APPLICATION
 	  Build as a native application that can run on the host and using
 	  resources and libraries provided by the host.
 
+config COMPILER_FREESTANDING
+	bool "Build in a freestanding compiler mode"
```

Comment:
Is there a reason we need this to be user selectable?

Reply:
I don't think so -- it's really just a consequence of an environment which is lacking a standard C library.

Reply:
So lets make this just `bool` and drop the user selectable nature

Reply:
User selection may still be desirable since dropping `-ffreestanding` enables a few optimisations using the libc functions, which user may or may not want.

Reply:
I think it best to wait for someone to come ask for this to be user selectable.  99.99% of users are not going to know how to set it and I don't think we should expose it until someone comes with a real need.

Reply:
> I think it best to wait for someone to come ask for this to be user selectable. 99.99% of users are not going to know how to set it and I don't think we should expose it until someone comes with a real need.

Why? Based on that reasoning, we should not have anything user configurable until someone comes with a need.

And I can already see the usefulness of having this user configurable (e.g. debugging a "mysterious" struct assignment failure to see whether it is due to the compiler making use of the libc `memcpy` by toggling this config in the menuconfig).

Reply:
As long as the defaults are correct, having this configurable by the user seems reasonable, it's something they "shouldn't" have to set, but it's also not something that Zephyr requires to be set in any particular way. This will also allow people to experiment with this flag, which might help them become more comfortable working in a 'hosted' environment.